src/flamegpu/io/Telemetry.cpp

```diff
@@ -0,0 +1,167 @@
+    telemetry_data.replace(telemetry_data.find(var_payload), var_payload.length(), payload);
+    // murder the formatting into a single compact line
+    telemetry_data.erase(std::remove(telemetry_data.begin(), telemetry_data.end(), '\n'), telemetry_data.end());  // Remove newlines and replace with space
+    telemetry_data.erase(std::remove(telemetry_data.begin(), telemetry_data.end(), '\t'), telemetry_data.end());  // Remove tabs and replace with space
+    telemetry_data.erase(std::remove_if(telemetry_data.begin(), telemetry_data.end(), [](char c) {
+            return std::isspace(static_cast<unsigned char>(c));
+        }), telemetry_data.end());  // Remove spaces
+    size_t pos = 0;
+    while ((pos = telemetry_data.find("\"", pos)) != std::string::npos) {   // Use escape characters
+        telemetry_data.replace(pos, 1, "\\\"");
+        pos += 2;
+    }
+
+    return telemetry_data;
+}
+
+bool sendTelemetryData(std::string telemetry_data) {
+    // Silent curl command (-s) and redirect response output to null
+    std::string null;
+#if _WIN32
+    null = "nul";
+#else
+    null = "/dev/null";
+#endif
+    std::string curl_command = "curl -s -o " + null + " -X POST \"" + std::string(TELEMETRY_ENDPOINT) + "\" -H \"Content-Type: application/json; charset=utf-8\" --data-raw \"" + telemetry_data + "\"";
+
+    // capture the return value
+    if (std::system(curl_command.c_str()) != EXIT_SUCCESS)
+        return false;
+
+    return true;
+}
+
+void hintTelemetryUsage() {
+    float random_normalised = static_cast<float>(std::rand()) / static_cast<float>(RAND_MAX);
+    // Dont print telemetry hints in test environment
+    if ((random_normalised < PROBABILITY_TELEMETERY_HINT) && !(flamegpu::util::isTestEnvironment()) && !(flamegpu::util::hasEnvironmentVariable("FLAMEGPU_SILENCE_TELEMETRY_NOTICE"))) {
+        fprintf(stdout, "NOTICE: The FLAME GPU software is reliant on evidence to support is continued development. Please "
+                        "consider enabling SHARE_USAGE_STATISTICS in your CMake options or calling 'shareUsageStatistics()' on "
+                        "your Simulation or Ensemble object. This message is randomly (p=%f) displayed but can be silenced "
+                        "using '--quiet' or by setting 'SILENCE_TELEMETRY_NOTICE' in your environment.\n", PROBABILITY_TELEMETERY_HINT);
+    }
+}
+
+bool silenceTelemetryNotice() {
+    return flamegpu::util::setEnvironmentVariable("FLAMEGPU_SILENCE_TELEMETRY_NOTICE", "True");
+}
+
+bool globalTelemetryEnabled() {
+    bool cmake_shared_usage_stats = false;
+#ifdef FLAMEGPU_SHARE_USAGE_STATISTICS
+    cmake_shared_usage_stats = true;
+#endif
+    std::string env_val_str = flamegpu::util::getEnvironmentVariable("FLAMEGPU_SHARE_USAGE_STATISTICS");
+    // Return false if explicitly disabled
+    if ((env_val_str == "0") || (env_val_str == "False") || (env_val_str == "FALSE") || (env_val_str == "Off") || (env_val_str == "OFF"))
```

Comment:
Given this value is also parsed by CMake for the default value of an `option`, it should match CMake false-yness conditions.

Couls also toupper/tolower it to reduce the checking for `False/false/FALSE` etc. 

https://cmake.org/cmake/help/latest/command/if.html#basic-expressions